### PR TITLE
[Merged by Bors] - chore(SetTheory/Ordinal/Arithmetic): deprecate `Ordinal.mex` and `Ordinal.bmex`

### PR DESCRIPTION
**Pull request by vihdzp**
There isn't any reason not to use `sInf sᶜ` instead - both nimber arithmetic and Grundy values work perfectly without `mex`.

The one nontrivial result was that the `mex` of an indexed family is less than the successor of the cardinal of the indexing type - this has been reproved as `sInf_compl_lt_ord_succ` and `sInf_compl_lt_ord_succ_lift`.

---

- [x] depends on: #16988
- [x] depends on: #16929

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/Arithmetic.lean [hunk 1]
```diff
@@ -1387,6 +1387,22 @@ theorem iSup_ord {ι} {f : ι → Cardinal} (hf : BddAbove (range f)) :
   conv_lhs => change range (ord ∘ f)
   rw [range_comp]
 
+theorem sInf_compl_lt_ord_succ_lift {ι : Type u} (f : ι → Ordinal.{max u v}) :
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Ordinal/Arithmetic.lean:1390]:
```suggestion
theorem sInf_compl_lt_lift_ord_succ {ι : Type u} (f : ι → Ordinal.{max u v}) :
```

**Comment by YaelDillies** [comment]
maintainer merge

apart from that

**Review by jcommelin — COMMENTED** [review]
bors d+

**Pull request closed without merging**